### ISSUE TITLE
Service_bus_queue Python

### ISSUE THREAD

**Issue by divyaspec** [Azure/azure-sdk-for-python#430]
Hi,
I am trying to create a Service Bus queue using python (PVTS tools on visual studio 2013).
<pre>
"""python
def create_service_bus_queue_simple(guid)
bus_service = ServiceBusService(service_namespace='SERVICE_BUS_NAMESPACE',shared_access_key_name='SHAREDACCESSKEYNAME',shared_access_key_value='SHAREDACCESSKEY')
    queue_options = Queue()
    queue_options.max_size_in_megabytes = '5120'
    queue_options.default_message_time_to_live = 'PT1M'
    bus_service.create_queue('jsonservicebusqueue', queue_options)
</pre>
But there is some bug and I am not sure where it is and how I can go about fixing the code. Can anyone have a look at it and help me to debug.

Please see below screenshot. I am using curl to post a JSON to my api which will upload the JSON as a blob and enable the service bus queue process as next step. This is where i am stuck and I tried to debug several times, no luck.
<img width="647" alt="screenshot" src="https://cloud.githubusercontent.com/assets/7250796/9064171/341f2962-3ac3-11e5-8433-fb5b1b9bde01.png">

**Comment by huguesv** [comment]
I get the same error as you if I forget to substitute in actual values for SERVICE_BUS_NAMESPACE, SHAREDACCESSKEYNAME and SHAREDACCESSKEY.

I can't share my actual key but I made one up.  It should look like this:
```
NS = 'mytesting1' # don't put the .servicebus.windows.net here, it's just the name
KEY_NAME = 'RootManageSharedAccessKey'
KEY_VAL = 'LTOs890jB8oLuarFbZV+N21G98GxGMg6bbYNKRzGjmY='
    bus_service = ServiceBusService(service_namespace=NS,shared_access_key_name=KEY_NAME,shared_access_key_value=KEY_VAL)
```

**Comment by divyaspec** [comment]
Hi, thank you for your comment. I actually have configured the values globally. For example I am using Flask framework so my code looks like this
<pre>
app.config['SERVICE_BUS_NAMESPACE'] = 'example-service-bus'
app.config['SHAREDACCESSKEYNAME'] = 'RootManageSharedAccessKey' 
app.config['SHAREDACCESSKEY'] = 'xxxxxxxxxxxxxxxxxxxxxxxxxxx'
</pre>

**Comment by huguesv** [comment]
There are different types of service bus namespaces, maybe you don't have the right one?
This is how I created mine:
![servicebus1](https://cloud.githubusercontent.com/assets/1696845/9072721/ad17d6fa-3ab1-11e5-900c-3b224d998de3.png)
Click on Create from the bottom toolbar
![servicebus2](https://cloud.githubusercontent.com/assets/1696845/9072725/b4e143a8-3ab1-11e5-9b8a-575ce83e8911.png)

**Comment by huguesv** [comment]
If I have the wrong key name or key value, I get an InvalidSignature error, so that's unlikely to be your issue.  If I have the wrong namespace, I get the getaddrinfo failed error like you.  So there must be something wrong with the namespace you are using (maybe wrong type like I suggest above).

**Comment by divyaspec** [comment]
I created exactly like you and I have the similar namespace. I am not sure why i am still getting an error. Another option will be to delete the current namespace and create a new one.

**Comment by huguesv** [comment]
Try creating one in the same region as me (West US), in case it's a problem isolated to your data center.

**Comment by divyaspec** [comment]
I am getting the same error & also, I created in the West US. I am not able to figure it out why this happens. I tried either ways by inserting values directly still the same issue.

**Comment by divyaspec** [comment]
I've started using azure queues and this works fine. I have no issues. Ideally I wanted to use service bus queue option to integrate to a different application service.

**Comment by huguesv** [comment]
Yeah, that's puzzling.  At that point, what I would do is try to create a service bus queue from another language, like node.js.  If that also fails, it's a sign of a bigger issue.  If not, then comparing the HTTP traffic between python and node.js can help.

At least you got it working with a storage queue.

**Comment by divyaspec** [comment]
Finally it worked, I had to create a queue on the portal and configure it by setting a new policy called listener and then select listen. Then it works. Now I can send and retrieve the msgs from the service_bus_queue. It was so weird because I followed the documentation on service bus [https://azure.microsoft.com/en-gb/documentation/articles/service-bus-python-how-to-use-queues/].
Thank you for your support though.
